test/test_kernel_replies.cpp

```diff
@@ -0,0 +1,116 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille and Sylvain Corlay                     *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include <iostream>
+
+#include "nlohmann/json.hpp"
+
+#include "test_interpreter.hpp"
+
+#include "xeus/xmessage_builder.hpp"
+#include "xeus/xguid.hpp"
+
+namespace nl = nlohmann;
+
+namespace test_kernel
+{
+
+    void test_interpreter::configure_impl()
+    {
+        auto handle_comm_opened = [](xeus::xcomm&& comm, const xeus::xmessage&) {
+            std::cout << "Comm opened for target: " << comm.target().name() << std::endl;
+        };
+        comm_manager().register_comm_target("test_target", handle_comm_opened);
+        using function_type = std::function<void(xeus::xcomm&&, const xeus::xmessage&)>;
+    }
+
+    nl::json test_interpreter::execute_request_impl(int execution_counter,
+                                                    const std::string& code,
+                                                    bool /* silent */,
+                                                    bool /* store_history */,
+                                                    nl::json /* user_expressions */,
+                                                    bool /* allow_stdin */)
+    {
+        nl::json kernel_res;
+
+        if (code.compare("hello, world") == 0)
+        {
+            publish_stream("stdout", code);
+        }
+
+        if (code.compare("error") == 0)
+        {
+            publish_stream("stderr", code);
+        }
+
+        if (code.compare("?") == 0)
+        {
+            std::string html_content = R"(<iframe class="xpyt-iframe-pager" src="
+                https://xeus.readthedocs.io"></iframe>)";
+            nl::json kernel_res;
+            kernel_res["payload"] = nl::json::array();
+            kernel_res["payload"][0] = nl::json::object({
+                {"data", {
+                    {"text/plain", "https://xeus.readthedocs.io"},
+                    {"text/html", html_content}}
+                },
+                {"source", "page"},
+                {"start", 0}
+            });
+            return xeus::create_successful_reply("", kernel_res["payload"]);
```

Comment:
There's something wrong with the way I send the `payload` to the jupyter_kernel_test.
I couldn't figure out what is the response that it is expecting nor the exact output of my payload code.
Can you shine a light on this @martinRenou?
Lemme know if I can explain this better to you, I can try!
Thank you.